### ISSUE TITLE
[v6 rc3][Bug] valid/invalid props are not updated

### ISSUE THREAD

**Issue by CarsonF** [erikras/redux-form#1344]
It seems like the `valid` & `invalid` props aren't updated in my component.

**Comment by garrettn** [comment]
I think I'm seeing the same thing. If I inspect the connected form component, I can see that there is an error in the `syncErrors` object, but `invalid` is false.

**Comment by garrettn** [comment]
After looking into my issue some more, and with some help from [this comment](https://github.com/erikras/redux-form/issues/1460#issuecomment-237560588), I discovered that the fields aren't always being registered in the store, so the `syncErrors` don't match up with the fields that are visible on the screen.

We're doing a wizard form of sorts, but it's different from the example in the docs in that it renders fields dynamically based on data from the server. The frontend doesn't know ahead of time what the form will look like, so it uses generic components for everything. As a result, sometimes the same Field components are shared between steps in the wizard, so their `componentWillMount` method doesn't get called, and that's where the field is supposed to be registered.

I found a way to work around the problem by rendering a loading spinner between steps until the step data is ready, so that the field components are re-mounted every time, but I was wondering if there's a better way that I'm missing.
